### ISSUE TITLE
cli: Environment variable values are printed into logs, potentially leaking sensitive info

### ISSUE THREAD

**Issue by a-robinson** [cockroachdb/cockroach#14168]
As part of supporting the zip command for packaging up debug info in https://github.com/cockroachdb/cockroach/pull/14076, we started printing out the values of environment variables in the logs. We had before made a conscious choice to only print the keys, because the values may include sensitive information like access tokens.

We definitely don't want to print the sensitive values in the logs, although I'm less sure about whether we want this in zips. I assume users wouldn't want them included, but some of the other env vars may be very relevant. Maybe we need a way of indicating which env vars are sensitive and only excluding their values? Or take this as motivation to move away from env vars sooner rather than later?

@petermattis

**Comment by petermattis** [comment]
We only print the COCKROACH* and GO* env var values to logs, right? Which of those are sensitive?

**Comment by a-robinson** [comment]
There's at least COCKROACH_LIGHTSTEP_TOKEN

**Comment by petermattis** [comment]
How sensitive is that? It doesn't give you access to lightstep data.

**Comment by a-robinson** [comment]
So I checked through the code and that does look like the only potentially sensitive one. Perhaps it's fine to print it if it the token only gives write access, or to do a one-off check to exclude that particular key's value. I just know I met some resistance from @tschottdorf when I made a similar change last year (https://github.com/cockroachdb/cockroach/issues/9136), and that we'll have to be careful with how things like S3/GCS credentials are provided for backup/restore.
